src/middlewares/requestValidator.js

```diff
@@ -0,0 +1,22 @@
+const createHttpError = require('http-errors');
+
+const requestValidator = function (schema) {
+  return async function (req, res, next) {
+    const method = req.method;
+    const dataSource = method === 'GET' ? 'query' : 'body';
+
+    try {
+      req[dataSource] = await schema.validateAsync(req[dataSource]);
```

Comment:
I don't think we need to modify req to accomplish this task. All you need to do is call the function

```js
await schema.validateAsync(req[dataSource]);
```

Reply:
I thought about it, and after validation we have true/false values casted to bool, it looks easier to work with

Reply:
the controller doesn't need to know anything about validators
the controller by default believes that the data is valid if the execution of the code reaches the controller